crates/schema/src/composed_schema.rs

```diff
@@ -326,15 +602,78 @@ impl ComposedSchema {
                                 meta_type.fields.insert(meta_field.name.clone(), meta_field);
                             }
                         } else {
+                            // Check if the type is marked as shareable before converting
+                            let mut type_is_shareable = false;
+                            for directive in &type_definition.node.directives {
+                                if directive.node.name.node.as_str() == "shareable" {
+                                    type_is_shareable = true;
+                                    break;
+                                }
+                            }
+
                             let meta_type = convert_type_definition(type_definition.node);
                             if let Some(meta_type2) = composed_schema.types.get(&meta_type.name) {
+                                let both_are_scalars =
+                                    meta_type.kind == TypeKind::Scalar && meta_type2.kind == TypeKind::Scalar;
+
+                                let common_scalar_types = ["DateTime", "Date", "Time", "JSON", "UUID", "Email", "URL"];
```

Comment:
Would be nice to put them on top as const.

Reply:
I also think that maybe it worth additional documentation on what are "common" scalars are and why GraphGate allows them to be defined in multiple subgraphs. If they will be a const on module level, it's a great place to put Rust Docs on.